lib/galaxy_test/api/test_jobs.py

```diff
@@ -243,9 +244,9 @@ def test_report_error_anon(self):
         job_id = run_response['jobs'][0]["id"]
         dataset_id = run_response['outputs'][0]['id']
         response = requests.post('%s/jobs/%s/error' % (self.galaxy_interactor.api_url, job_id),
-                                 params={'email': 'someone@domain.com', 'dataset_id': dataset_id},
+                                 data={'email': 'someone@domain.com', 'dataset_id': dataset_id},
                                  cookies=cookies)
-        assert response.status_code == 200
+        assert response.status_code == 200, response.json()
```

Comment:
```suggestion
        assert response.status_code == 200, response.text
```
It's very likely that non-200 status codes won't be valid JSON.

Reply:
This worked fine in my testing:
```
self = <galaxy_test.api.test_jobs.JobsApiTestCase testMethod=test_report_error_anon>

    @skip_without_tool('detect_errors_aggressive')
    def test_report_error_anon(self):
        # Need to get a cookie and use that for anonymous tool runs
        cookies = requests.get(self.url).cookies
        payload = json.dumps({"tool_id": "detect_errors_aggressive",
                              "inputs": {"error_bool": "true"}})
        run_response = requests.post("%s/tools" % self.galaxy_interactor.api_url, data=payload, cookies=cookies).json()
        job_id = run_response['jobs'][0]["id"]
        dataset_id = run_response['outputs'][0]['id']
        response = requests.post('%s/jobs/%s/error' % (self.galaxy_interactor.api_url, job_id),
                                 data={'email': 'someone@domain.com', 'dataset_id': dataset_id},
                                 cookies=cookies)
>       assert response.status_code == 200, response.text
E       AssertionError: {"err_msg": "Job is not in error", "err_code": 400008}
E       assert 400 == 200
E         -400
E         +200

lib/galaxy_test/api/test_jobs.py:248: AssertionError
```

and is used already in `lib/galaxy_test/api/test_search.py` and `lib/galaxy_test/api/test_tools.py`

Reply:
Make it a 500 and it won't work.

Reply:
And we should change those other instances too. It's a minor thing, but it isn't quite right.